samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala

```diff
@@ -34,15 +33,14 @@ class PetApi(baseUrl: String) {
    * 
    * @param pet Pet object that needs to be added to the store
    */
-  def addPet(pet: Pet): RequestT[Identity, Either[ResponseException[ErrorModel, Exception], Pet], Any] =
+  def addPet(pet: Pet): Request[Either[ResponseException[ApiModel, Exception], Pet], Any] =
     basicRequest
       .method(Method.POST, uri"$baseUrl/pet")
       .contentType("application/json")
       .body(pet)
       .response(fromMetadata(
-        asJsonEither[ErrorModel, Pet],
-        ConditionalResponseAs(_.code == StatusCode(200), asJsonEither[ErrorModel, Pet]),
-        ConditionalResponseAs(_.code == StatusCode(405), asJsonEither[MethodNotAllowed, Pet])
```

Comment:
It will be nice to have all statuses mapped to Error Models rather than have one generic ApiModel. Otherwise we loose specification provided by OpenApi and have to implement it outside of autogenerated client.

Ideally it will be nice to convert model into exception with ErrorModel supported, unfortunately OpenApi doesn't support it directly.

here the example what i mean. Something like this:
```scala
case class PetModelNotAllowed[ErrorModel](result: ErrorModel) extends Exception

client.addPet(myPetRequest) match {
 case Left(successfulPetModel) => doSomething(successfulPetModel)
 case Right(ex: PetModelNotAllowed, _) =>
   logger.error(s"Oops, something went wrong. Code for reference ${ex.errorCode}. ${ex.errorMessage}
   case Right(_, ex: Throwable) =>
     logger.error("Request failed", ex)
 }
```

Reply:
Even generic `ApiException` can be fine to handle ErrorModel 
```scala
case class ApiException[T: ErrorModel](statusCode: HttpStatus, errorModel: T) extends Exception

// optional:
case class PetModelNotAllowed extends ApiException[ErrorModel]
```
but the main point is to have httpCode to Model mapping specified in api specification as it provided by OpenApiSpec

Reply:
Thanks for your input, however you seem to be reviewing the original commit. In 77025d8 I got rid of `ErrorModel`, because after all it doesn't make any sense - there is no official model for error, so now it will only handle the error content if it's explicitly specified in the spec, for example:
```yml
responses:
  '200':
    description: successful operation
    content:
      application/xml:
        schema:
          $ref: '#/components/schemas/Pet'
      application/json:
        schema:
          $ref: '#/components/schemas/Pet'
  '404':
    description: Pet not found
    content:
      application/xml:
        schema:
          $ref: '#/components/schemas/Error'
      application/json:
        schema:
          $ref: '#/components/schemas/Error'
```